frontend/projects/upgrade/src/app/shared/services/common-dialog.service.ts

```diff
@@ -251,7 +251,45 @@ export class DialogService {
         sourceList,
         sourceAppContext: appContext,
         action: UPSERT_PRIVATE_SEGMENT_LIST_ACTION.EDIT_FLAG_EXCLUDE_LIST,
-        flagId: flagId,
+        id: flagId,
+      },
+    };
+    return this.openUpsertPrivateSegmentListModal(commonModalConfig);
+  }
+
+  openAddListModal(appContext: string, segmentId: string) {
```

Comment:
When these modals are opened, it calls the legacy action to fetch segments, which deletes the member values for the lists (since they are subsegments).